common/include/serial_port.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * serial_port.h
+ *
+ *  Created on: 2020��9��18��
+ *      Author: duanning
+ */
+
+#ifndef INCLUDE_SERIAL_PORT_H_
+#define INCLUDE_SERIAL_PORT_H_
+
+#include <iostream>
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include "abstract_device.h"
+#include "abstract_protocol.h"
+
+using namespace boost::asio;
+
+class SerialPort: public AbstractDevice {
+public:
+	enum character_size_type {character_size_5 = 5, character_size_6 = 6, \
+		character_size_7 = 7, character_size_8 = 8};
+	enum parity_type {parity_none = 0, parity_odd = 1, parity_even = 2};
+	enum stop_bits_type {stop_bits_one, stop_bits_onepointfive, stop_bits_two};
+	enum flow_control_type {flow_control_none, flow_control_software, flow_control_hardware};
+
+    static const constexpr char* kSerialPortName 	= "serial_port_name";
```

Comment:
Please pay attention to white spaces, you're mixing spaces and tabs, we are required to use tab only. Normally we should turn on the "show whitespace characters", it's a good habit.

Reply:
Thank you. I will pay attention to this problem.